state/src/storage/tx/raft_log.rs

```diff
@@ -230,11 +230,19 @@ impl<'db> StateTxn<'db, RW> {
     /// Purge all logs before the given index (inclusive)
     pub fn purge_logs(&self, index: u64) -> Result<(), StorageError> {
         let mut log_cursor = self.logs_cursor()?;
-
-        let idx_lsn = lsn_to_key(index);
-        log_cursor.seek(&idx_lsn)?;
-        while log_cursor.has_current()? {
-            log_cursor.delete()?;
```

Comment:
I see, was the issue here that when deleting, the cursor's position is changed (whereas we thought it would stay the same)?

Reply:
no delete was implemented correctly. The issue is that the method deletes the wrong set of values